### PR TITLE
enable and fix doctests for IPython.utils

### PR DESCRIPTION
**Pull request by ivanov**
As I mentioned in #10021, these doctests weren't enabled by default, so I am enabling them and fixing the one test that's currently broken.

### CODE REVIEW

File: IPython/utils/text.py [hunk 1]
```diff
@@ -639,7 +639,7 @@ def _find_optimal(rlist, row_first=False, separator_size=2, displaywidth=80):
         if sumlength + separator_size * (ncols - 1) <= displaywidth:
             break
     return {'num_columns': ncols,
-            'optimal_separator_width': (displaywidth - sumlength) / (ncols - 1) if (ncols - 1) else 0,
+            'optimal_separator_width': (displaywidth - sumlength) // (ncols - 1) if (ncols - 1) else 0,
```
Comment (ivanov) [inline IPython/utils/text.py]: this resulted in an optimal_separator_width of e.g. 2.5 on the doctest example for `compute_item_matrix` below which is probably not what was intended.

**Comment by ivanov** [comment]
LoL @minrk being able to delete my branches 😹   Thanks, Min!

**Comment by minrk** [comment]
@ivanov sorry! I habitually delete my own merged-PR branches when I see that button, and sometimes miss when it's someone else who's given me commit rights to their repo. I hope I didn't overstep too much.

**Comment by ivanov** [comment]
not at all, @minrk, it's awesome, thank you! I don't always make it back to GH to delete the branch myself, so that helps.

**Pull request merged**